test/release/examples/primers/loops.chpl

```diff
@@ -0,0 +1,950 @@
+
+/*
+
+  Because each iteration is executed by a separate task,
+  coforall-loops can synchronize between distinct iterations of the
+  loop, unlike the data-parallel loop forms.  This makes
+  coforall-loops useful as a means of creating an arbitrary number of
+  tasks that are independently doing the same, or similar, things.  In
+  practice, the parallel iterators used to define ``forall`` loops are
+  often implemented in terms of coforall-loops that create a task per
+  processor core and/or locale.  As an example, the following
+  coforall-loop creates a task per processor core across all locales:
+
+*/
+
+
+  coforall loc in Locales {
+    on loc {
+      coforall tid in 1..numTasks {
+        /* This loop body will be executed once per core per locale */
+      }
+    }
+  }
+
+/*
+
+  For further information about tasks in Chapel, see the :ref:`Task
+  Parallelism Primer <primers-taskParallel>`.
+
+  When using coforall loops, keep in mind that a task will be created
+  for each iteration, and that each task will consume memory and
+  processing resources.  For example, a coforall loop with a million
+  iterations will literally create a million tasks.  If you don't have
+  a million cores to run them on, this is likely to be overkill,
+  requiring more memory and processing power than may be warranted.
+  If there is no explicit synchronization between the iterations,
+  perhaps a forall loop would be a better choice, since it would use a
+  number of tasks proportional to the targeted hardware parallelism?
+
+
+  Closing Discussions
+  ===================
+
+  At this point, you've learned about all of Chapel's loop forms.  The
+  remaining sections cover some loop-related discussion topics that
+  come up frequently in practice.
+
+
+  Nesting Loops
+  -------------
+
+  The loop forms discussed here can be nested arbitrarily, and their
+  definitions are the same whether they are an outer or inner loop.  A
+  nested for-loop will perform nested serial iterations as in other
+  languages.  A nested coforall-loop will create a number of tasks
+  equal to outer loop's trip count and the sum of all the inner loops'
+  counts.  For example, the following loop will logically create
+  ~`x**2` tasks, since each iteration of both loops will create its
+  own task.
+
+*/
+
+  config const x = 4;
+
+  writeln();
+  coforall i in 1..x do
+    coforall j in 1..x do
+      writeln("Here's a message from one of the nested coforall tasks");
+
+/*
+
+  A tricky case to reason about in a nested loop situation is the
+  forall-loop since its implementation is essentially "Do whatever the
+  parallel iterator says to."  If a parallel iterator were to always
+  create ``x`` tasks (say), then a nested forall loop that invoked
+  that iterator for both loops would create ``x**2`` tasks as in the
+  coforall example above.
+
+  In practice, however, most parallel iterators (including those
```

Comment:
It seems like this information on nested loops is more important than zippered iteration.  Thus going along with my suggestion to move the zippered iteration piece later, I recommend moving it after this information.